### PR TITLE
Implement Source integration tests

### PR DESCRIPTION
**Pull request by alyakimenko**
### Description

This PR seeks to implement Source integration tests.

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/conduitio-labs/conduit-connector-hubspot/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -111,4 +111,14 @@ issues:
         - funlen # it can be annoying for table-driven tests
         - dupl
         - goerr113
-        - maintidx
\ No newline at end of file
+        - maintidx
+    - path: _integration_test
```
Comment (voscob) [inline .golangci.yml:115]: 🥇

File: source/source_integration_test.go [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+	record, err := source.Read(ctx)
+	is.NoErr(err)
+
+	is.Equal(record.Operation, sdk.OperationSnapshot)
+
+	testContactID := compareTestContactWithRecord(t, is, testContact, record)
+	err = hubspotClient.Delete(ctx, testResource, testContactID)
+	is.NoErr(err)
+
+	err = source.Teardown(ctx)
```
Comment (voscob) [inline source/source_integration_test.go:92]: Shouldn't we close the context above and call the `Teardown` method with the new one? 🤔

File: source/source_integration_test.go [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+	err := hubspotClient.Update(ctx, testResource, itemID, updatedTestContact)
+	if err != nil {
+		return nil, fmt.Errorf("update contact: %w", err)
+	}
+
+	return updatedTestContact, nil
+}
+
+// waitTestContacts waits until the expected count of contacts will be presented in HubSpot.
+func waitTestContacts(
```
Comment (voscob) [inline source/source_integration_test.go:338]: Do we really need this method?
IIUC, do you want to wait for all records to be written in the specified order before calling the method?

**Review by voscob — COMMENTED** [review]
Do we need a test case where one more record is added after the Snapshot starts, so we can be sure it will be read by the CDC (not by the Snapshot)?

File: source/source_integration_test.go (hunk 3 above)
Comment (alyakimenko) [reply source/source_integration_test.go]: Yeah, it seems that I forgot to remove it after adding the `readWithRetry` function. Thanks for the catch!

File: source/source_integration_test.go (hunk 2 above)
Comment (alyakimenko) [reply source/source_integration_test.go]: Yeah, this definitely will be better. 👍

**Review by voscob — APPROVED** [review]
@alyakimenko yep, my bad, you're right.

Approved 👍

**Pull request merged**